### ISSUE TITLE
Weird code/bug in ImageFont when loading font..

### ISSUE THREAD

**Issue by andrew-davie** [python-pillow/Pillow#7127]
Mmh. I see the code on my system is totally different to the github code for the ImageFont file -- and the code on github actually does do the handling of the font load completely differently from the code I show below. So it's probably already a fixed bug and the real issue is why the hell does my version of Pillow say it's up to date but link to completely different/old code.  But that's another isssue so I'm closing this.  :(

I am using Pillow on MacOS M2 Ventura 13.0
My Pillow library is 9.5.0 and I'm running PyCharm.
Python version is 3.9

I have struck a crash/bug when attempting to load a truetype font with `font = ImageFont.truetype("Arial.ttf", 24)`
I am importing via `from PIL import ImageFont`
Here is the dump/trace of the error...

``` 
Traceback (most recent call last):
File "/Users/boo/Documents/software/something/venv/lib/python3.9/site-packages/PIL/ImageFont.py", line 982, in truetype
    return freetype(font)
  File "/Users/boo/Documents/software/something/venv/lib/python3.9/site-packages/PIL/ImageFont.py", line 979, in freetype
    return FreeTypeFont(font, size, index, encoding, layout_engine)
  File "/Users/boo/Documents/software/something/venv/lib/python3.9/site-packages/PIL/ImageFont.py", line 238, in __init__
    load_from_bytes(font)
  File "/Users/boo/Documents/software/something/venv/lib/python3.9/site-packages/PIL/ImageFont.py", line 232, in load_from_bytes
    self.font_bytes = f.read()
AttributeError: 'str' object has no attribute 'read'
```

The code in question (ImageFont.py about line #231) is shown here...
```
       def load_from_bytes(f):
            self.font_bytes = f.read()
            self.font = core.getfont(
                "", size, index, encoding, self.font_bytes, layout_engine
            )

        if is_path(font):
            load_from_bytes(font)
```

The call I made from my code was...
```
               font = ImageFont.truetype("Arial.ttf", 24)
```
I know what the bug is; I just can't really understand how this has escaped into the wild so far.
Essentially, stepping into that code, it gets to ImageFont.py line #238 where it goes 'load_from_bytes(font)' where font is a string. And then load_from_bytes tries to f.read() from that string -- which of course totally fails, because f should be a file at that point, not a string.  

I fixed this locally - rightly or wrongly - by changing line 238 to 'load_from_bytes(open(font, 'rb'))' -- and for me, at least, loading fonts now correctly works; I see the expected font being loaded and I don't get the error dump.

I find this very very strange as I would expect this error to be happening a LOT to everyone.  So I must be doing something wrong, but I can't see it and the obvious "fix" I have made does actually make things work. Strange.

Here's a full two-liner that shows the bug on my machine. And it works once I modify that line in ImageFont.py mentioned above.

```
from PIL import ImageFont
font = ImageFont.truetype("Arial.ttf", 24)
```

**Comment by andrew-davie** [comment]
Update to this - I uninstalled Pillow in the PyCharm IDE and instead installed Pillow-PIL (which installed Pillow too). I also installed latest Pillow via pip on my machine. I undersatnd PyCharm has everyting inside a virtual env so the install on my system should not have mattered. But in any case now, stepping into the code, I see the latest code and of course the bug doesn't happen.
Leaving this comment here to help others who might have the same issue.
